source/analysis/AMAdditionAB.cpp

```diff
@@ -104,6 +104,15 @@ bool AMAdditionAB::axisValues(int axisNumber, int startIndex, int endIndex, doub
 void AMAdditionAB::onInputSourceValuesChanged(const AMnDIndex& start, const AMnDIndex& end)
 {
 	cacheUpdateRequired_ = true;
+
+	AMnDIndex scanStart = start;
+	AMnDIndex scanEnd = end;
+	scanStart.setRank(start.rank()-1);
+	scanEnd.setRank(end.rank()-1);
+
+	if (scanStart == scanEnd)
+		dirtyIndices_ << start;
+
```

Comment:
At the moment, there are only two ways analysis blocks update.  It is either one value at a time (step scanning) or full repaint (from size change, source change, etc.).